docs/community/20230104-mixed-clusters.md

```diff
@@ -0,0 +1,52 @@
+---
+title: Feature Group for Mixed Clusters in Cluster API
+authors:
+  - "@prometherion"
+  - "@lentzi90"
+reviewers:
+  - "@lentzi90"
+  - "@richardcase"
+creation-date: 2023-01-04
+last-updated: 2023-01-06
+status: proposed
+see-also:
+  - https://github.com/kubernetes-sigs/cluster-api/issues/7475#issuecomment-1302454745
+  - https://kubernetes.slack.com/archives/C8TSNPY4T/p1667458956589639
+---
+# Mixed Clusters in Cluster API Feature Group
+
+This document briefly outlines the scope, communication media, and stakeholders for a formal Feature Group dedicated to defining a Cluster API-approved solution for "mixed clusters" which combines two or more infrastructure providers.
+
+## User Story and Problem Statement
+
+As a Kubernetes cluster administrator, I want the ability to specify different infrastructure providers for the Control Plane and the Worker nodes to decouple each-other infrastructure providers.
+
+At the current state, Cluster API is giving for granted that worker nodes and control plane components are backed by the same infrastructure provider. However, this is a hard statement, since the two components could be backed by different infrastructure providers.
```

Comment:
I think we should be a little bit more specific here.

This is not a Cluster API limitation, nothing in Cluster API prevents the deployment of mixed clusters (neither in the API nor in the contract), and providers like Kubemark already works today in a "mixed clusters" scenario.

Some of the Cluster API providers instead were developed assuming clusters to exist in a single infrastructure provider, mostly because each infrastructure brings in a set of concerns shared by all the machines running on the same infrastructure, like networking, network credentials, failure domains, and probably more.

Given that this was an accepted and consistent starting point all the provider's team agreed on and worked hard to achieve, I also suggest phrasing the goal of this working group as an initiative for supporting a new use case, e.g.

"We aim to extend and build on the great work done so far by providers, making it possible to use different infrastructure providers in the same cluster as well as ensuring a consistent UX when using different combinations of providers"

Reply:
Very well put!
If you agree @prometherion , please use the quote @fabriziopandini provided instead of line 24.